packages/gatsby-source-filesystem/src/create-file-node.js

```diff
@@ -35,7 +36,21 @@ exports.createFileNode = async (
       description: `Directory "${path.relative(process.cwd(), slashed)}"`,
     }
   } else {
-    const contentDigest = await md5File(slashedFile.absolutePath)
+    const key = stats.mtimeMs.toString() + stats.ino.toString()
+    let contentDigest
+
+    if (pluginOptions.fastHash) {
+      // Skip hashing.
+      contentDigest = key
+    } else {
+      // Generate a hash, but only if the file has changed.
+      contentDigest = cache && (await cache.get(key))
+      if (!contentDigest) {
+        contentDigest = await md5File(slashedFile.absolutePath)
```

Comment:
My PR https://github.com/gatsbyjs/gatsby/pull/37433 got merged, so you can rebase onto master and use the md5File function from there now.

```js
const { md5File } = require(`gatsby-core-utils`)
```

The `md5-file` package can be uninstalled then :)